tests/jballerina-unit-test/src/test/java/org/ballerinalang/test/query/QueryExprWithQueryConstructTypeTest.java

```diff
@@ -204,6 +204,43 @@ public void testNegativeScenarios() {
                 "incompatible types: expected '(Department & readonly)', found 'Department'", 283, 55);
         validateError(negativeResult, index++, "incompatible types: expected '[string,string]', " +
                 "found '([string,int]|[string,int]|[string,int]|[string,int])'", 286, 48);
+        validateError(negativeResult, index++, "incompatible types: expected 'map<(int[2] & readonly)> & readonly'," +
+                " found '((map<(int[2] & readonly)> & readonly)|error)'", 292, 34);
+        validateError(negativeResult, index++, "incompatible types: expected '(Department & readonly)'," +
+                " found 'Department'", 297, 55);
+        validateError(negativeResult, index++, "incompatible types: expected '[string,string]', " +
```

Comment:
Here, why are we getting the collection type?

Reply:
It seems it's the same behavior as `foreach`. Type resolved for item is `[string, int]|[string, int]|[string, int]|[string, int]`
```ballerina
public function main() {
    foreach var item in [["1", 1], ["2", 2], ["3", 3], ["4", 4]] {
        string _ = item; //incompatible types: expected 'string', found '([string,int]|[string,int]|[string,int]|[string,int])
    }
}

```

Reply:
Yeah. Therefore, we don't need to improve the error msg here right?